ethereum_history_api/circuits/lib/src/misc/uint256.nr

```diff
@@ -0,0 +1,50 @@
+use dep::std::ops::Add;
+
+global uint128_overflow_value = 340282366920938463463374607431768211456; // 2^128
+
+struct U256 {
+    high: U128,
+    low: U128,
+}
+
+impl U256 {
+    fn new(high: U128, low: U128) -> Self {
+        Self { high, low }
+    }
+
+    fn zero() -> Self {
+        Self { high: U128::from_integer(0), low: U128::from_integer(0) }
+    }
+
+    fn one() -> Self {
+        Self { high: U128::from_integer(0), low: U128::from_integer(1) }
+    }
+}
+
+impl Eq for U256 {
+    fn eq(self, other: U256) -> bool {
+        (self.high == other.high) & (self.low == other.low)
+    }
+}
+
+impl Add for U256 {
+    fn add(one: Self, other: Self) -> Self {
+        let lo: Field = one.low.to_integer() + other.low.to_integer();
+
+        let mut low = U128::from_integer(0);
+        let mut carry = 0;
+        if(lo.lt(uint128_overflow_value)){
```

Comment:
Why not use `<`? I think it's a bit clearer
```suggestion
        if(lo < uint128_overflow_value){
```

Reply:
Generally Fields cannot be compared. Fortunately there is this special method `lt`, which allows just this one comparison, but it cannot be made using `<` sign. And we need Fields here.